### DISCUSSION TITLE
cant connect from asus router

### DISCUSSION THREAD

**Discussion by Array-zer0** [pivpn/pivpn#1668] [Support]
Trying to connect to my home router with an asus ax3000 routers built in openvpn client.
I am able to connect to my PIA vpn with no issues.
I have tried connecting to my pivpn connections and none have worked.
I tried my "work" profile, which i can connect to from the openvpn client in windows with no issues
if i try the work profile (passworded) it fails and the log shows the following error

Dec 22 10:00:18 vpnclient4[9249]: neither stdin nor stderr are a tty device and you have neither a controlling tty nor systemd - can't ask for 'Enter Private Key Password:'.  If you used --daemon, you need to use --askpass to make passphrase-protected keys work, and you can not use --auth-nocache.
Dec 22 10:00:18 vpnclient4[9249]: Exiting due to fatal error

after some research it looks like the -nopass option should resolve this but when I do that, i get

Dec 22 10:02:53 vpnclient2[9492]: VERIFY X509NAME ERROR: CN=pivpn_d1981d25-5499-4aa6-845b-73c306######, must be pivpn_d1981d25-5499-4aa6-845b-7
Dec 22 10:02:53 vpnclient2[9492]: OpenSSL: error:1416F086:lib(20):func(367):reason(134)
Dec 22 10:02:53 vpnclient2[9492]: TLS_ERROR: BIO read tls_read_plaintext error
Dec 22 10:02:53 vpnclient2[9492]: TLS Error: TLS object -> incoming plaintext read error
Dec 22 10:02:53 vpnclient2[9492]: TLS Error: TLS handshake failed
Dec 22 10:02:53 vpnclient2[9492]: SIGUSR1[soft,tls-error] received, process restarting
Dec 22 10:02:53 vpnclient2[9492]: Restart pause, 20 second(s)

my .ovpn file shows

tls-version-min 1.2
verify-x509-name pivpn_d1981d25-5499-4aa6-845b-73c306###### name
cipher AES-256-CBC

any help would be greatly appreciated, im working at a remote site for the next few months, and most of my equipment needs to be able to see the servers on the other side

**Comment by DominikPalo** [comment]
I'm experiencing the same issue - I own the ASUS RT-AC68U router with the most recent version of ASUSWRT (3.0.0.4.386_51665). I generated a passwordless OVPN profile using the `pivpn add nopass` command. Then I imported that profile to my ASUS router, which is installed on a different network/site and tried to connect. Initially, I got an error:
```
Dec 23 20:20:53 vpnclient5[14682]: neither stdin nor stderr are a tty device and you have neither a controlling tty nor systemd - can't ask for 'Enter Private Key Password:'.  If you used --daemon, you need to use --askpass to make passphrase-protected keys work, and you can not use --auth-nocache.
```

So, I removed the `auth-nocache` line from the OVPN config, reimported it and now I'm getting a new error:

```
Dec 23 20:30:16 vpnclient5[16650]: VERIFY EKU OK
Dec 23 20:30:16 vpnclient5[16650]: VERIFY X509NAME ERROR: CN=piaware_f023718e-af87-43cb-ac24-27fcb2edd6ea, must be piaware_f023718e-af87-43cb-ac24
Dec 23 20:30:16 vpnclient5[16650]: OpenSSL: error:1416F086:lib(20):func(367):reason(134)
Dec 23 20:30:16 vpnclient5[16650]: TLS_ERROR: BIO read tls_read_plaintext error
Dec 23 20:30:16 vpnclient5[16650]: TLS Error: TLS object -> incoming plaintext read error
Dec 23 20:30:16 vpnclient5[16650]: TLS Error: TLS handshake failed
```

So, for some reason the CN from the generated certificate doesn't match the required value/name - not sure why. Any ideas?

BTW, the PiVPN server configuration seems fine, as I'm able to connect and use that VPN from different devices (macOS and Windows computer) without any issues - only the ASUS router refuses to connect.

**Comment by DominikPalo** [comment]
UPDATE: According to [this discussion](https://www.snbforums.com/threads/openvpn-verify_x509_name-char-array-size.81614/), the ASUSWRT implementation limits X509 NAME to a maximum of 31 characters. However, the NAME generated by PiVPN is in my case 44 characters long (`{hostname}_{uuid}`). This is why it gets truncated on the router side and doesn't match the CN stored in the certificate. 😞

As I workaround, I **removed the verify-x509-name ... line** from the OVPN config file and now it works fine.
